airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -407,3 +407,18 @@ def get_external_log_url(self, task_instance: TaskInstance, try_number: int) ->
     def supports_external_link(self) -> bool:
         """Whether we can support external links"""
         return bool(self.frontend)
+
+
+def safe_attrgetter(*items, obj, default):
+    """
+    Get items from obj but return default if not found
+
+    :meta private:
+    """
+    val = None
+    try:
+        val = attrgetter(*items)(obj)
+    except AttributeError:
+        pass
+
+    return val or default
```

Comment:
I’d name this something like `getattr_nested`

Also defauling to None does not feel like a good idea. You’ll want NOTSET. Or it’s also a good idea to just inline this logic in `_group_logs_by_host`

Reply:
it does not default to None -- it defaults to _default_, which is required!

Reply:
or that's the intention lemme confirm

Reply:
yup! lmkyt

Reply:
Sorry, I meant the `val or default` part, if the attrgetter gets a None, the function would return `default` instead of `None`, inconsistent to `getattr`.

Reply:
ok renamed

Reply:
slash updated